types/protractor-http-mock/index.d.ts

```diff
@@ -11,38 +12,11 @@ declare namespace mock {
         /**
          * Instantiate mock module. This must be done before the browser connects.
          *
-         * @param mocks An array of mock modules to load into the application.
-         * @param plugins An array of Plugin objects.
+         * @param mocks An array of either mock modules or module names relative to the rootDirectory configuration to load into the application.
+         * @param plugins An array of either Plugin objects or NPM modules as strings.
          * @param skipDefaults Set true to skip loading of default mocks.
          */
-        <TResponse, TPayload>(mocks?: Array<requests.AllRequests<TResponse, TPayload>>, plugins?: Array<Plugin>, skipDefaults?: boolean): ProtractorHttpMock;
-
-        /**
-         * Instantiate mock module. This must be done before the browser connects.
-         *
-         * @param mocks An array of mock modules to load into the application.
-         * @param plugins An array of NPM modules as strings.
-         * @param skipDefaults Set true to skip loading of default mocks.
-         */
-        <TResponse, TPayload>(mocks?: Array<requests.AllRequests<TResponse, TPayload>>, plugins?: Array<string>, skipDefaults?: boolean): ProtractorHttpMock;
-
-        /**
-         * Instantiate mock modules from files. This must be done before the browser connects.
-         *
-         * @param mocks An array of mock module names relative to the rootDirectory configuration.
-         * @param plugins An array of Plugin objects.
-         * @param skipDefaults Set true to skip loading of default mocks.
-         */
-        (mocks: Array<string>, plugins?: Array<Plugin>, skipDefaults?: boolean): ProtractorHttpMock;
-
-        /**
-         * Instantiate mock modules from files. This must be done before the browser connects.
-         *
-         * @param mocks An array of mock module names relative to the rootDirectory configuration.
-         * @param plugins An array of NPM modules as strings.
-         * @param skipDefaults Set true to skip loading of default mocks.
-         */
-        (mocks: Array<string>, plugins?: Array<string>, skipDefaults?: boolean): ProtractorHttpMock;
+        (mocks?: ReadonlyArray<requests.AllRequests | string>, plugins?: ReadonlyArray<Plugin1<any> | Plugin2<any, any> | string>, skipDefaults?: boolean): ProtractorHttpMock;
```

Comment:
Currently the type of each mock and each plug-in is checked separately. Because of that all 4 overloaded methods were merged into a single one.

https://github.com/atecarlos/protractor-http-mock/blob/77df9c34a6cb837856fdb23e9cfb580a51ea4aec/lib/initData.js#L47
https://github.com/atecarlos/protractor-http-mock/blob/77df9c34a6cb837856fdb23e9cfb580a51ea4aec/lib/initData.js#L72

Reply:
As it is shown in the `TestCtorOverloads` test the function can receive mocks of different types, e.g. `Get<number>, Delete<string>`, hence `AllRequests` cannot point at one specific generic type.